Enforce LF Line Endings

ObliviousHarmony opened the pull request:
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This pull request uses a `.gitattributes` file to enforce LF line endings for all text files in the monorepo.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create branch from this one locally.
2. Create `test.js` with CLRF line endings and add a few lines.
3. Commit the file.
4. Checkout `trunk`.
5. Checkout your test branch.
6. Note that the line endings have changed to LF.

<!-- End testing instructions -->

ObliviousHarmony committed:
Forced LF Line Endings

Since it's possible that people may be cloning the repository in
Windows, we need to enforce consistent line endings.

ObliviousHarmony committed:
Changelog

adrianduffell commented on plugins/woocommerce/.gitattributes:
```diff
@@ -1,20 +0,0 @@
-# Declare files that will always have LF line endings on checkout.
-*.php text eol=lf
-
-# Remove files for archives generated using `git archive`.
-/.*                export-ignore
-bin                export-ignore
-CODE_OF_CONDUCT.md export-ignore
-changelog.txt      export-ignore
-composer.*         export-ignore
-Gruntfile.js       export-ignore
-package.json       export-ignore
-package-lock.json  export-ignore
-phpcs.xml          export-ignore
-phpunit.*          export-ignore
-README.md          export-ignore
-renovate.json      export-ignore
-tests              export-ignore
```
Is it safe to remove these `export-ignore` lines? Some tooling might use `git archive` under the hood.

ObliviousHarmony replied:
Given that it's possible, I've gone ahead and updated this file list to reflect the current state of the `plugins/woocommerce` directory.

ObliviousHarmony committed:
Excluded Files From Archive

The `export-ignore` files are used when building the ZIP for
distribution. This makes sure they match the files that should
be excluded.

adrianduffell reviewed (approved):
This tested great and LGTM! Thanks @ObliviousHarmony 🚀
